### PR TITLE
CORE-11648 Added message bus admin list topic feature

### PR DESCRIPTION
**Pull request by owenstanford**
Add an admin function to the message bus to support listing available topics. This is required for the external app messaging changes. We need to check topics exist before trying to send external messages.

### CODE REVIEW

File: libs/messaging/kafka-message-bus-impl/src/main/kotlin/net/corda/messagebus/kafka/admin/builder/KafkaAdminBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.osgi.service.component.annotations.Component
+
+@Suppress("Unused")
+@Component(service = [AdminBuilder::class])
+class KafkaAdminBuilder : AdminBuilder {
+    override fun createAdmin(adminConfig: AdminConfig, messageBusConfig: SmartConfig): Admin {
+        val configResolver = MessageBusConfigResolver(messageBusConfig.factory)
+        val kafkaProperties = configResolver.resolve(messageBusConfig, adminConfig)
+
+        val contextClassLoader = Thread.currentThread().contextClassLoader
```
Comment (vkolomeyko) [inline libs/messaging/kafka-message-bus-impl/src/main/kotlin/net/corda/messagebus/kafka/admin/builder/KafkaAdminBuilder.kt]: There is a utility function `executeWithThreadContextClassLoader` to simplify this operation.

File: libs/messaging/message-bus/src/main/kotlin/net/corda/messagebus/api/admin/builder/AdminBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import net.corda.messagebus.api.configuration.AdminConfig
+import net.corda.messagebus.api.admin.Admin
+
+/**
+ * Builder Interface for creating instances of [Admin].
+ */
+interface AdminBuilder {
+
+    /**
+     * Generate kafka producer with given properties.
```
Comment (vkolomeyko) [inline libs/messaging/message-bus/src/main/kotlin/net/corda/messagebus/api/admin/builder/AdminBuilder.kt]:
```suggestion
     * Generate a message bus admin with given properties.
```

File: libs/messaging/message-bus/src/main/kotlin/net/corda/messagebus/api/configuration/AdminConfig.kt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package net.corda.messagebus.api.configuration
+
+data class AdminConfig(val clientId: String)
```
Comment (vkolomeyko) [inline libs/messaging/message-bus/src/main/kotlin/net/corda/messagebus/api/configuration/AdminConfig.kt:3]: Should we put some notes about uniqueness of `clientId`?

File: libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/persistence/DBAccess.kt [hunk 4]
```diff
@@ -107,6 +107,16 @@ class DBAccess(
         }
     }
 
+    fun getAllTopics():Set<String>{
```
Comment (vkolomeyko) [inline libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/persistence/DBAccess.kt]: Nit:
```suggestion
    fun getAllTopics(): Set<String>{
```

**Review by vkolomeyko — COMMENTED** [review]
LGTM

Few minor comments posted

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by mcgovc — APPROVED** [review]
LGTM - build changes

**Pull request merged**